pkg/slo-controller/noderesource/resource_calculator.go

```diff
@@ -71,33 +73,85 @@ func (r *NodeResourceReconciler) updateNodeResource(node *corev1.Node, nr *frame
 	strategy := sloconfig.GetNodeColocationStrategy(r.cfgCache.GetCfgCopy(), node)
 
 	r.prepareNodeResource(strategy, nodeCopy, nr)
+	needSyncStatus, needSyncMeta := r.isNodeResourceSyncNeeded(strategy, node, nodeCopy)
+	if !needSyncStatus && !needSyncMeta {
+		klog.V(5).InfoS("skip update node resource for node", "node", node.Name)
+		return nil
+	}
+
+	var errList []error
+	if needSyncStatus {
+		if err := util.RetryOnConflictOrTooManyRequests(func() error {
+			return r.updateNodeStatus(node, strategy, nr)
+		}); err != nil {
+			klog.ErrorS(err, "failed to update node status for node", "node", node.Name)
+			errList = append(errList, err)
+		}
+	}
 
-	if needSync := r.isNodeResourceSyncNeeded(strategy, node, nodeCopy); !needSync {
+	if needSyncMeta {
+		if err := util.RetryOnConflictOrTooManyRequests(func() error {
```

Comment:
If the status has been updated once before, there will be a conflict for the first time here. Do you need to update the data in advance?

Reply:
Yes. In the `updateNodeStatus()` and the `updateNodeMeta`, we Get the node object via the controller-runtime client first and then update the status/metadata of the node.
It should avoid most of the conflicts, while for some others (since the controller-runtime client reads from the informer cache), we can simply retry.